src/reader.py

```diff
@@ -0,0 +1,177 @@
+    split_train_and_test(list_email: list[str], list_label: list[int])
+        return tuple of train and test data
+    """
+
+    def __init__(self, patch):
+        self.patch = patch
+        self.dataset = DSReader.load_dataset(self.patch)
+        self.dictionary = []
+
+    @staticmethod
+    def load_dataset(path):
+        """Method for loading a dataset
+        Parameters
+        ----------
+        path : str
+            full path to the dataset
+        """
+
+        data_frame_csv = pd.read_csv(path)
+        return data_frame_csv
+
+    def remove_digits(self):
+        """Method for remove digits from a dataset
+        """
+
+        def no_digits(inp_str):
+            inp_str = str(inp_str)
+            no_digits_str = ''
+            for char in inp_str:
+                if char not in string.digits:
+                    no_digits_str = no_digits_str + char
+                else:
+                    no_digits_str = no_digits_str + ' '
+            return no_digits_str
+
+        self.dataset['email'] = self.dataset['email'].map(no_digits)
+        return self.dataset
+
+    def to_lower(self):
+        """Method for converting a dataset to lowercase
+        """
+
+        self.dataset['email'] = self.dataset['email'].str.lower()
+        return self.dataset
+
+    def remove_punctuation_marks(self):
+        """Method for remove punctuation marks from a dataset
+        """
+
+        def no_punctuation(inp_str):
+            inp_str = str(inp_str)
+            no_punctuation_str = ""
+            for char in inp_str:
+                if char not in string.punctuation:
+                    no_punctuation_str = no_punctuation_str + char
+                else:
+                    no_punctuation_str = no_punctuation_str + ' '
+            return no_punctuation_str
+
+        self.dataset['email'] = self.dataset['email'].map(no_punctuation)
+        return self.dataset
+
+    def remove_stopwords(self):
+        """Method for remove stopwords from a dataset
+        """
+
+        stop_words = tuple(nltk.corpus.stopwords.words('english'))
+
+        def no_stopwords(inp_str):
+            words = nltk.word_tokenize(inp_str)
+            new_list_str = ' '.join(words)
+            words = (new_list_str.lower()).split()
+
+            without_stop_words = [word.lower() for word in words
+                                  if word not in stop_words]
+            without_short_words = [x for x in without_stop_words
+                                   if len(x) > 2]
+            new_str = ' '.join(without_short_words)
+            return new_str
+
+        self.dataset['email'] = self.dataset['email'].map(no_stopwords)
+        return self.dataset
+
+    def remove_duplicates(self):
+        """Method for remove duplicates from a dataset
+        """
+
+        self.dataset = pd.DataFrame.drop_duplicates(self.dataset)
+        return self.dataset
+
+    def make_dictionary(self):
```

Comment:
This method should create a list of words from the whole dataset but each word should be unique - e.g. only one occurrence for each word. Looks like task description on the grow portal is not quite clear. The **list** in the description does not mean that it should be a **list** data type - in fact it should be just a ordered array of words (and in this case you could use any convenient data type).

Reply:
Done:
  - fixed the make_dictionary method,
  - added function logging,
  - minor corrections in the operation of other methods.